### DISCUSSION TITLE
Device object name, location, description lengths

### DISCUSSION THREAD

**Discussion by mykdor** [bacnet-stack/bacnet-stack#675] [Q&A] [answered]
There are macros in device.h:

```C
#define MAX_DEV_NAME_LEN 32
#define MAX_DEV_LOC_LEN  64
#define MAX_DEV_MOD_LEN  32
#define MAX_DEV_VER_LEN  16
#define MAX_DEV_DESC_LEN 64
```
Are there restrictions regarding the length of, say,  the device object name... or the description in the standard? I havn't find any. If so, can I safely redefine those?

**Comment by skarg** [comment, marked as answer]
Yes, you can change those to whatever values suits your application.  I think they were only relevant for the gateway example application.

In the config.h file, there are some limits to the size of a characterstring (internal representation) which is based on size of the APDU that the device supports:
```
#ifndef MAX_CHARACTER_STRING_BYTES
#define MAX_CHARACTER_STRING_BYTES (MAX_APDU-6)
#endif
```
All the datalinks support an APDU of 50 bytes (min), and BACnet MS/TP and ARCNET support up to 480 bytes (max), and extended MS/TP and BACnet/IP and BACnet/IPv6 and BACnet Ethernet and BACnet/SC all support up to 1476 bytes APDU.

[BTL Implementation Guidelines](https://btl.org/wp-content/uploads/sites/3/2022/08/btl_implementation_guideline-1.pdf) suggest:
> 7.14 Writable Character String properties should support strings of useful length
Writable Character String properties should support strings of useful length. Although “useful” is a vague
term, an Object_Name property that can contain at most a four octet Character String is not likely to be useful
to a user, whereas an Object_Name of length 512 is likely to be far more than adequate.

**Reply by mykdor** [reply]
Thanks a lot!
